api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubIssueV3.java

```diff
@@ -0,0 +1,138 @@
+
+        if (!(issueObject instanceof Map)) {
+            return "Issue Data Not Found";
+        }
+
+        Map issueMap = (Map) issueObject;
+
+        Object repoMap = jsonObject.get("repository");
+        if (repoMap == null) { return "Repository Data Not Available"; }
+
+        String repoUrl = restClient.getString(repoMap, "html_url");
+        GitHubParsed gitHubParsed = new GitHubParsed(repoUrl);
+
+        // Picking the "default_branch" here, didn't find specific branch information on the github webhook issue payload.
+        // Also, an issue is seen at the repo-level across all the branches on github, it is not attached a specific branch.
+        String branch = restClient.getString(repoMap, "default_branch");
+
+        GitRequest issue = getIssue(issueMap, gitHubParsed, branch);
+
+        if (issue != null) { gitRequestRepository.save(issue); }
+
+        return result;
+    }
+
+    protected GitRequest getIssue(Map issueMap, GitHubParsed gitHubParsed, String branch) {
+        if (issueMap.isEmpty()) { return null; }
+
+        GitRequest issue = new GitRequest();
+        issue.setRequestType("issue");
+        issue.setScmUrl(gitHubParsed.getUrl());
+        issue.setScmBranch(branch);
+        issue.setOrgName(gitHubParsed.getOrgName());
+        issue.setRepoName(gitHubParsed.getRepoName());
+        issue.setTimestamp(System.currentTimeMillis());
+
+        String number = restClient.getString(issueMap, "number");
+        issue.setScmRevisionNumber(number);
+        issue.setNumber(number);
+
+        String message = restClient.getString(issueMap, "title");
+        issue.setScmCommitLog(message);
+
+        Object userObject = issueMap.get("user");
+        String name = restClient.getString(userObject, "login");
+        issue.setUserId(name);
+
+        String created = restClient.getString(issueMap, "created_at");
+        long createdTimestamp = new DateTime(created).getMillis();
+        issue.setCreatedAt(createdTimestamp);
+
+        String updated = restClient.getString(issueMap, "updated_at");
+        long updatedTimestamp = new DateTime(updated).getMillis();
+        issue.setUpdatedAt(updatedTimestamp);
+
+        issue.setClosedAt(0);
+        issue.setResolutiontime(0);
+        issue.setMergedAt(0);
+        issue.setState("open");
+        String state = restClient.getString(issueMap, "state");
+        if ("CLOSED".equalsIgnoreCase(state)) {
+            String closed = restClient.getString(issueMap, "closed_at");
+            long closedTimestamp = new DateTime(closed).getMillis();
+            issue.setScmCommitTimestamp(closedTimestamp);
+            issue.setClosedAt(closedTimestamp);
+            issue.setMergedAt(closedTimestamp);
+            issue.setResolutiontime((closedTimestamp - createdTimestamp));
+            issue.setState("closed");
+        }
+
+        setCollectorItemId(issue);
+
+        return issue;
+    }
+
+    protected void setCollectorItemId(GitRequest issue) {
```

Comment:
see comment on GitHubCommitV3 for this method

Reply:
Same reply thoughts as on GitHubCommitV3, thanks